.claude/commands/analyze-ticket.md

```diff
@@ -0,0 +1,91 @@
+---
+description: Deeply analyze a Linear ticket before implementation — pass a ticket ID or leave blank to let Claude infer it
+---
+
+<!--
+If $ARGUMENTS is blank, infer the ticket ID from context.
+If it can't be inferred, ask the user to supply one and stop.
+-->
+
+Ultrathink about the comprehensive discovery needed before implementation. What context, risks, and unknowns must be uncovered to ensure a successful implementation that passes review on the first attempt?
+
+## 0. Prepare Fresh Environment
+
+Ensure you're starting from a clean, up-to-date state:
+
+```bash
+git checkout main
+git pull origin main
+```
+
+Analyze **$ARGUMENTS** deeply:
+
+## 1. Requirements Analysis
+
+- Parse the ticket description and acceptance criteria
+- **Compare ticket requirements against current codebase state**
+- **Is the ticket outdated? Have requirements changed since creation?**
+- **Do any requirements need modification based on current implementation?**
+- Identify explicit and implicit requirements
+- Clarify any ambiguities or missing information
+- Define success metrics and validation criteria
+
+## 2. Technical Discovery
+
+- Review existing code and architecture patterns
+- Identify files and components that need modification
+- Check for similar implementations to follow as patterns
+- Assess technical complexity and estimated effort
+
+## 3. Risk Assessment
+
+- What could go wrong during implementation?
+- Are there backward compatibility concerns?
+- Performance implications to consider?
+- Security considerations?
+- What edge cases need handling?
+
+## 4. Dependencies & Blockers
+
+- Check for dependencies on other tickets or systems
+- Verify all necessary access and permissions
+- Identify any potential blockers
+- Confirm prerequisite work is completed
+
+## 5. Documentation Review
+
+- Review existing documentation for context
+- Identify documentation that will need updates:
```

Comment:
Maybe we just add something to keep the documentation as small and clear as possible?